### PR TITLE
DSS Infra: Chalice IAM Policy

### PR DESCRIPTION
**Pull request by amarjandu**
Removes different IAM policies depending on deployment source.
Only updates IAM policies from non-CI environments (operator machines).
Addresses malformed IAM policy for dss-{stage} lambda. 

Connected To #2681 , #1973 , #1955

### CODE REVIEW

File: iam/policy-templates/dss-lambda.json [hunk 1]
```diff
@@ -69,7 +69,6 @@
         "Action": [
             "dynamodb:*"
         ],
-        "Resource": "arn:aws:dynamodb:*:$account_id:table/dss-async-state-$stage",
```
Comment (xbrianh) [inline iam/policy-templates/dss-lambda.json]: Consider making this change in a separate PR to disentangle deploy script changes from IAM policy updates.

File: chalice/build_deploy_config.sh [hunk 2]
```diff
@@ -60,11 +60,14 @@ for var in $(echo $env_json | jq -r keys[]); do
...
+account_id=$(aws sts get-caller-identity | jq -r .Account)
+export iam_role_arn="arn:aws:iam::${account_id}:role/dss-${stage}"
+cat "$config_json" | jq .manage_iam_role=false | jq .iam_role_arn=env.iam_role_arn | sponge "$config_json"
 cat "$iam_policy_template" | envsubst '$DSS_S3_BUCKET $DSS_S3_CHECKOUT_BUCKET $dss_es_domain $account_id $stage' > "$policy_json"
 cp "$policy_json" "$stage_policy_json"
+
+if [[ ${CI:-} != true ]]; then
+	# IAM policies must be updated from an operators machine, this will not run on CI environments.
+	echo "updating $iam_role_arn with $stage_policy_json"
+	aws iam put-role-policy --role-name $lambda_name --policy-name $lambda_name --policy-document file://$stage_policy_json
```
Comment (xbrianh) [inline chalice/build_deploy_config.sh]: What's the advantage of this over `cat "$config_json" | jq .manage_iam_role=true`?

File: iam/policy-templates/dss-lambda.json (hunk 1 above)
Comment (amarjandu) [reply iam/policy-templates/dss-lambda.json]: shall do, ill merge it in first, as it needs to be fixed for the syntax to work in `aws  iam put-role-policy`

File: chalice/build_deploy_config.sh (hunk 2 above)
Comment (amarjandu) [reply chalice/build_deploy_config.sh]: For a given stage `dev`;
we dont use `auto-gen-policies` in our deployments script, by setting `manage_iam_role:true` we would end up with a policy that is called `dss-dev-function_handler`; there is no problem with this, the only issue is that when the CI uploads its manually managed IAM policy it uses `policy-dev.json`, but calls uses the arn name `dss-{dev}`. 

By always having a manually managed IAM Role we can keep using the same name `dss-dev` but only update the policy when we are not in a CI system. 

:)

File: iam/policy-templates/dss-lambda.json (hunk 1 above)
Comment (amarjandu) [reply iam/policy-templates/dss-lambda.json]: @xbrianh please see https://github.com/HumanCellAtlas/data-store/pull/2685

**Pull request merged**